tools/querytee/proxy_endpoint_test.go

```diff
@@ -335,7 +335,7 @@ func Test_ProxyEndpoint_SummaryMetrics(t *testing.T) {
 			require.Equal(t, uint64(tc.counts), requestCount.Load())
 
 			require.Eventually(t, func() bool {
-				return prom_testutil.ToFloat64(proxyMetrics.responsesComparedTotal) == 1
+				return prom_testutil.CollectAndCount(proxyMetrics.responsesComparedTotal) == 1
```

Comment:
When I read your PR description, I was expecting to find that you added a `require.Eventually`, but now I you changed how we were evaluating inside the eventually. can you quickly explain the difference?